seed/energy/meter_data_processor/monthly_data_aggregator.py

```diff
@@ -0,0 +1,105 @@
+import datetime
+import json
+import logging
+
+import requests
+from dateutil import tz
+from django.conf import settings
+
+from seed.models import (
+    Meter,
+    TimeSeries,
+)
+
+
+
+_log = logging.getLogger(__name__)
+
+
+def aggr_sum_metric(data, localtzone):
+    '''
+    aggregate monthly data kairos and push it to postgres. data represents the aggregate query
+    '''
+
+    headers = {'content-type': 'application/json'}
+
+    url = settings.TSDB['query_url']
+
+    data = json.dumps(data)
+
+    r = requests.post(url, data=data, headers=headers)
+    # length of output array. Should be 1 per group since it's monthly aggregation and we are querying only for one month
+
+    if not 'queries' in r.json():
+        return
+
+    res = r.json()['queries'][0]['results']
+
+    length = len(res)
+
+    # Retrieve required values from array and call posgres insert function
+    for num in range(0, length):
```

Comment:
Thanks, Piper. I will update the code as you suggested.

On Thu, Feb 11, 2016 at 11:59 AM, Piper Merriam <notifications@github.com>
wrote:

> In seed/energy/meter_data_processor/monthly_data_aggregator.py
> <https://github.com/SEED-platform/seed/pull/714#discussion_r52631174>:
>
> > +    url = settings.TSDB['query_url']
> > +
> > +    data = json.dumps(data)
> > +
> > +    r = requests.post(url, data=data, headers=headers)
> > +    # length of output array. Should be 1 per group since it's monthly aggregation and we are querying only for one month
> > +
> > +    if not 'queries' in r.json():
> > +        return
> > +
> > +    res = r.json()['queries'][0]['results']
> > +
> > +    length = len(res)
> > +
> > +    # Retrieve required values from array and call posgres insert function
> > +    for num in range(0, length):
>
> You can do this style of iteration like the following.
>
> for idx, value in enumerate(result):
>     ....
> # These two are the same.
> for idx in range(len(res)):
>     value = res[idx]
>     ...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SEED-platform/seed/pull/714/files#r52631174>.
>